### ISSUE TITLE
Add datalog example to RacketScript playground?

### ISSUE THREAD

**Issue by spdegabrielle** [racketscript/racketscript#268]
```
#lang datalog

ancestor(A, B) :- parent(A, B).
ancestor(A, B) :-
  parent(A, C), ancestor(C, B).
parent(john, douglas).
parent(bob, john).
ancestor(A, B)?
```

**Comment by spdegabrielle** [comment]
I saw @samphilips demo of compiling rhombus to RacketScript and I was wondering if examples in other `#lang`’s are possible?

**Comment by stchang** [comment]
Good idea. I can't get it to work because we don't support imaginary numbers at the moment. Though I'm not sure why datalog needs them. Will look into it.

**Comment by sorawee** [comment]
They are from the contract library -- `racket/collects/racket/contract/private/generate-base.rkt`.

Supposing this problem is fixed, there's still another problem at run-time: `TypeError: M3.hash_for_each is not a function` in the `racket/pretty`'s compiled module.

**Comment by stchang** [comment]
`hash-for-each` should be easy enough to add.

**Comment by stchang** [comment]
I'm also seeing a failure on call to `make-semaphore`. Does anyone know if contracts are multi-threaded?
